src/Plutus/Certification/Internal.hs

```diff
@@ -48,3 +53,18 @@ splitString maxChars = toValue . chunksOf maxChars
 split64 :: Text -> Value
 split64 = splitString 64
 
+type WithDB = forall a m. (MonadIO m, MonadMask m) => (forall n. (DB.MonadSelda n,MonadMask n) => n a) -> m a
+
+newtype WithDBWrapper = WithDBWrapper WithDB
+
+class HasDb env where
+  getWithDb :: env -> WithDB
+
+instance HasDb WithDBWrapper where
+  getWithDb (WithDBWrapper db') = db'
+
+withDb :: (MonadReader env m,HasDb env, MonadIO m,MonadMask m) => (forall n. (DB.MonadSelda n,MonadMask n) => n a) -> m a
+withDb dbAction = do
+  env <- ask
+  getWithDb env dbAction
+
```

Comment:
- `executeDbAction env dbAction == getWithDb env dbAction` therefore no need of it .
- though `ask >>= flip getWithDb dbAction` looks good to me is not compiling because of compilation error 
`• Could not deduce (DB.MonadSelda n0)
    arising from a use of ‘dbAction’
`